keras_nlp/src/utils/preset_utils.py

```diff
@@ -192,6 +200,43 @@ def get_file(preset, path):
         )
 
 
+def load_preset_from_gcs(fname, url, cache_subdir):
+    """Much of this is adapted from get_file of keras core."""
+    if url is None:
+        raise ValueError(
+            'Please specify the "url" argument (URL of the file '
+            "to download)."
+        )
+
+    if cache_subdir is None:
+        cache_dir = config.keras_home()
+
+    datadir_base = os.path.expanduser(cache_dir)
+    if not os.access(datadir_base, os.W_OK):
+        datadir_base = os.path.join("/tmp", ".keras")
+    datadir = os.path.join(datadir_base, cache_subdir)
+    os.makedirs(datadir, exist_ok=True)
+
+    fname = path_to_string(fname)
```

Comment:
I think everything after this line can be much simpler... something like

```python
src = os.path.join(remote_dir, filename)
dst = os.path.join(data_dir, filename)
if not os.path.exists(dst):
    tf.io.gfile.copy(src, dst)
return dst
```

Reply:
I wanted to retain many of the original error checks and corrections to prevent introducing a change that might break someone else's usage. 

That being said I did simplify a bit. Let me know what you think.

Reply:
Commented below, I think most of these branches are for things that make sense for an exposed `get_file` util, but not for an internal helper. We know we call this with all args supplied, let's write the function as such.